plaidml/bridge/openvino/ops/roi_pooling.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+#include "plaidml/op/op.h"
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+#include <tuple>
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace {
+
+template <typename T>
+std::vector<T> cast_constant_operand(size_t operand_idx, ngraph::Node* layer) {
+  auto* ngraph_const = ngraph::as_type<ngraph::op::Constant>(layer->get_input_node_ptr(operand_idx));
+  if (ngraph_const) {
+    return ngraph_const->cast_vector<T>();
+  } else {
+    THROW_IE_EXCEPTION << "Dynamic slicing not currently supported by PlaidML plugin; all of begin, end, and stride "
+                          "must be Constants.";
+  }
+}
+
+template <typename T>
+Buffer makeBuffer(DType dtype, const std::vector<int64_t>& dims, const std::vector<T>& data) {
+  TensorShape shape(dtype, dims);
+  Buffer buffer(shape);
+  buffer.copy_from(data.data());
+  return buffer;
+}
+
+template <typename T>
+edsl::Tensor indices_generator(T start_point, int64_t len, int stride, DType dtype, std::string name) {
+  std::vector<T> indices(len);
+  for (int64_t i = 0; i < len; i++) {
+    indices[i] = static_cast<T>(start_point + stride * i);
+  }
+  return edsl::Constant(makeBuffer(dtype, {len}, indices), name);
+}
+
+std::tuple<edsl::Tensor, int, int> crop_resized(edsl::Tensor& I, std::vector<float>& coord, std::string method,
+                                                int64_t pooled_h, int64_t pooled_w) {
+  auto x_1 = coord[0];
+  auto y_1 = coord[1];
+  auto x_2 = coord[2];
+  auto y_2 = coord[3];
+
+  auto roi_width = std::max(x_2 - x_1, 1.0f);
+  auto roi_height = std::max(y_2 - y_1, 1.0f);
+
+  int kernel_h = static_cast<int>(std::floor(roi_height / pooled_h));
+  int kernel_w = static_cast<int>(std::floor(roi_width / pooled_w));
+
+  edsl::Tensor w_tensor, h_tensor;
+  edsl::InterpolationMode interpolation_mode;
+  if (method != "bilinear") {
+    h_tensor = indices_generator(static_cast<int>(x_1), pooled_h, kernel_h, DType::INT32, "indices_h");
```

Comment:
Can you use edsl::index here?
please see 
https://github.com/plaidml/plaidml/pull/1548/files#diff-8683f58c8e11af7cc0957be604d50b7ca9f0e6ff336c45b49a2c6c9c0292d74eR68

Reply:
Done, replaced indices_generator by edsl::index op.